### PR TITLE
fix(oci): create blobs dir and define blobs dir name constant

### PR DESCRIPTION
**Pull request by sparr**
https://github.com/opencontainers/image-spec/blob/v1.0/image-layout.md#content specifies that the blobs dir must exist, so the changed ensureDir in this PR is a bugfix.

The addition and use of the ociBlobsDir constant is more of a cleanup suggestion, and I can remove it if necessary. I've also asked for it to be added upstream at https://github.com/opencontainers/image-spec/issues/1069 but that will probably take significantly longer.

### CODE REVIEW

**Review by qweeah — APPROVED** [review]
LGTM, but I am not a maintainer

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Comment by sparr** [comment]
Thank you both.

@TerryHowe, can you approve running the check workflows?

**Comment by sparr** [comment]
Created https://github.com/opencontainers/image-spec/pull/1071 which would make the new constant here unnecessary.

**Comment by sparr** [comment]
https://github.com/opencontainers/image-spec/pull/1071 is merged. If they have a release before this PR is merged, I'll switch over to using the blobs constant from there, ditto for ociImageIndexFile

**Comment by shizhMSFT** [comment]
> [opencontainers/image-spec#1071](https://github.com/opencontainers/image-spec/pull/1071) is merged. If they have a release before this PR is merged, I'll switch over to using the blobs constant from there, ditto for ociImageIndexFile

Cool. Let's wait for `image-spec v1.1.0-rc.4` with your commits in there.

**Review by shizhMSFT — APPROVED** [review]
LGTM. We may merge this PR if the next release of `image-spec` takes too long.

**Comment by sparr** [comment]
Sadly I missed the cutoff for rc4 by just a few hours. For unrelated reasons I've been talking to the maintainers over there about their releases, and it's unclear whether rc5 is days or months away, but weeks seems most likely.

**Pull request merged**
